File: Napakalaki_Java/src/napakalaki/Treasure.java
/**
 *  Marco Manuel Fernández Pranno, 2015 (Granada)
 *  Para la asignatura PDOO (Universidad de Granada)
 *  Licencia GPLv3 a partir del 15/02/16, hasta entonces todos los derechos reservados.
 */

package napakalaki;

public class Treasure {
    
    private String name;
    private int bonus;
    private TreasureKind type;
    
    public Treasure(String n, int bonus, TreasureKind t){
        this.name = n;
        this.bonus = bonus;
        this.type = t;
    }

    public String getName() {
        return name;
    }

    public int getBonus() {
        return bonus;
    }

    public TreasureKind getType() {
        return type;
    }
    
}

File: Napakalaki_Ruby/pkg/Napakalaki_Ruby-0.0.1/lib/treasure.rb
#encoding: utf-8

=begin
  Marco Manuel Fernández Pranno, 2015 (Granada)
  Para la asignatura PDOO (Universidad de Granada)
  Licencia GPLv3 a partir del 15/02/16, hasta entonces todos los derechos reservados.
=end 

class Treasure
  def initialize(name, bonus,tkind)
    @name = name
    @bonus = bonus
    @tkind = tkind
  end
  
  attr_reader :name,:bonus,:tkind
  
  def to_s
    "#{@name}(#{@bonus}-#{@tkind}) \n"
  end
  
end
